### DISCUSSION TITLE
Lua filter

### DISCUSSION THREAD

**Discussion by esigo** [envoyproxy/gateway#2852] [Q&A] [answered]
Is it possible to use Lua filter in httpRoute? If yes, is there any example for it?
Thanks

**Comment by arkodg** [comment, marked as answer]
until a first class API is added (tracked with https://github.com/envoyproxy/gateway/issues/2025) you can use EnvoyPatchPolicy to configure one https://gateway.envoyproxy.io/latest/user/extensibility/envoy-patch-policy/

**Reply by esigo** [reply]
Thanks for the prompt response ☺️

**Reply by esigo** [reply]
@arkodg unfortunately I couldn't get a basic Lua filter working. Would it be possible to share a hello world example for Lua filter or let me know how I could get the following patch fixed?

```python
apiVersion: gateway.envoyproxy.io/v1alpha1
kind: EnvoyPatchPolicy
metadata:
  name: custom-lua-filter-patch-policy
  namespace: default
spec:
  targetRef:
    group: gateway.networking.k8s.io
    kind: Gateway
    name: eg
    namespace: default
  type: JSONPatch
  jsonPatches:
    - type: "type.googleapis.com/envoy.config.listener.v3.Listener"
      name: default/eg/http  # Replace if listener name format differs
      operation:
        op: add
        path: "/default_filter_chain/filters/-"  # Add filter at the end (-)
        value:
          name: envoy.lua
          config:
            inlineCode: |
              function envoy_on_response(response_handle)
                local headers = response_handle:headers()
                headers:add("custom-header", "value")
              end

              register_filter("envoy.on_response", envoy_on_response)

```

**Reply by arkodg** [reply]
if youre attaching to a https listener the `path` looks a little different  - `filterChains/0/filters`

**Comment by gauravitdevops** [comment]
Hi @arkodg 
I have been working on to configure Envoy Proxy Gateway. I am able to create the HTTPRoute and add SecurityPolicy and enable the JWT token validation in my current setup, the only challenge faced is that I am unable to create Lua filter to modify the headers. Can you help with some examples to use Lua script in Kind as EnvoyPatchPolicy. 
Expecting a quick turnaround.

Thanks,
Gaurav

**Reply by CR-FLDCDE** [reply]
I would be very glad about a example how to make use of lua.

**Comment by rsmartins78** [comment]
I have a similar scenario. I need to extract a field from a request body in JSON and add the value on a new request header (basically making the rate limit work properly on my OAuth endpoint), but I just don't see how I can do it with the Envoy gateway. I already have the Lua script and tested it on Envoy.

**Reply by zirain** [reply]
you can use EnvoyPatchPolicy as a workaround.

**Reply by rsmartins78** [reply]
I don't have much experience with envoy, do you have by any chance an example of how could I add the lua code to EnvoyPatchPolicy ?

**Comment by arkodg** [comment]
raised https://github.com/envoyproxy/gateway/issues/4627 so first class support can be added for Lua
